packages/pg-typed/src/index.ts

```diff
@@ -227,6 +228,7 @@ class Table<TRecord, TInsertParameters> {
   ) {
     this._tableID = tableName;
     this._value = (c, v) => serializeValue(c, v);
+    this.tableName = tableName.format(pgFormat).text;
```

Comment:
The `tableName` as a `string` cannot be used in queries in the way you suggest in your examples. It would get escaped as a string, rather than an identifier. I think it might work better to just replace `private readonly _tableID: SQLQuery;` with `public readonly tableId: SQLQuery;`.

If needed, users can then call `tableId.fromat(pgFormat).text` to the the actual table name.

Alternatively, we could expose just the raw table name (without the schema) as `public readonly tableName: string`, which you could then use via `sql.ident(tableName)` in other queries.